rfcs/2020-04-04-2221-kubernetes-integration.md

```diff
@@ -0,0 +1,1726 @@
+   and interesting tactics in their approach though.
+1. [Influx Helm charts]
+1. [Awesome Operators List] - an "awesome list" of operators.
+
+## Sales Pitch
+
+See [motivation](#motivation).
+
+## Drawbacks
+
+1. Increases the surface area that our team must manage.
+
+## Alternatives
+
+1. Not do this integration and rely solely on external community-driven
+   integrations.
+
+## Outstanding Questions
+
+### From Ben
+
+1. ~~What is the best to avoid Vector from ingesting it's own logs? I'm assuming
+   that my [`kubectl` tutorial][anchor_tutorial_kubectl] handles this with
+   namespaces?
+   We'd just need to configure Vector to exclude this namespace?~~
+   See the [Origin filtering][anchor_origin_filtering] section.
+1. ~~From what I understand, Vector requires the Kubernetes `watch` verb in order
+   to receive updates to k8s cluster changes. This is required for the
+   `kubernetes_pod_metadata` transform. Yet, Fluentbit [requires the `get`,
+   `list`, and `watch` verbs][fluentbit_role]. Why don't we require the same?~~
+   Right, this is a requirement since we're using k8s API. The exact set of
+   permissions is to be determined at YAML files design stage - after we
+   complete the implementation. It's really trivial to determine from a set of
+   API calls used.
+   See the [Deployment Hardening](#deployment-hardening) section.
+1. What are some of the details that set Vector's Kubernetes integration apart?
+   This is for marketing purposes and also helps us "raise the bar".
+
+### From Mike
+
+1. What significantly different k8s cluster "flavors" are there? Which ones do
+   we want to test against? Some clusters use `docker`, some use `CRI-O`,
+   [etc][container_runtimes]. Some even use [gVisor] or [Firecracker]. There
+   might be differences in how different container runtimes handle logs.
+1. How do we want to approach Helm Chart Repository management.
+1. How do we implement liveness, readiness and startup probes?
+   Readiness probe is a tricky one. See [Container probes](#container-probes).
+1. Can we populate file at `terminationMessagePath` with some meaningful
+   information when we exit or crash?
+1. Can we allow passing arbitrary fields from the `Pod` object to the event?
+   Currently we only to pass `pod_id`, pod `annotations` and pod `labels`.
+
+## Plan Of Attack
+
+- [x] Agree on minimal Kubernetes version. (1.14)
+- [x] Agree on a list of Kubernetes cluster flavors we want to test against.
+- [ ] Setup a proper testing suite for k8s.
+  - [ ] Local testing via `make test-integration-kubernetes`.
+    - [ ] Ability to "bring your own cluster". See [issue#2170].
+  - [ ] Add `make test-integration-kubernetes` to the `ci.yaml` workflow.
+    - [ ] Ensure these tests are stable. See [issue#2193], [issue#2216],
+          and [issue#1635].
+    - [ ] Ensure we are testing all supported minor versions. See
+          [issue#2223].
+  - [ ] Run `make test-integration-kubernetes` against AWS' EKS platform in
```

Comment:
@MOZGIII let's discuss this on Monday. I think _just_ AWS EKS is sufficient for now.

Reply:
I agree, we can add other sources later.